### ISSUE TITLE
Make Internet Explorer check compatible with uglify

### ISSUE THREAD

**Issue by renholm** [rodneyrehm/viewport-units-buggyfill#43]
The isOldInternetExplorer variable is always false if using uglify (and maybe other minification methods) since comments gets stripped. I suggest changing to another method for checking Internet Explorer versions.

**Comment by rodneyrehm** [comment]
sounds reasonable. PR welcome!

**Comment by dillonlucente** [comment]
We ran into a similar issue here with the updated version of this library. Uglify was moving the @cc_on comments out from where they originally existed in the code, so it executes those checks outside of the initialization block. It's too late to assign variables, even if they did exist in the global namespace. 

Since there's some user agent sniffing going on for iOS and Android, we just moved the checking for buggy and old IEs to the same format, specifically:

```js
  var isBuggyIE = /MSIE [0-9]\./i.test(userAgent);
  var isOldIE = /MSIE [0-8]\./i.test(userAgent);
```

The regex is a bit hacky, but at least the idea should be sound, unless I'm missing something that you ran into. I guess you could do some more checking, like MS does [here](https://msdn.microsoft.com/en-us/library/ms537509%28v=VS.85%29.aspx).

This might close #40 as well. If this sounds good, I'd be happy to do a PR.

Great library, by the way, saved us days of headaches.

**Comment by rodneyrehm** [comment]
looks good to me. any objections from @zoltan-dulac (who originally authored this part)? feel free to send a PR :)

**Comment by rodneyrehm** [comment]
thank you for the fix, it's released as [v0.5.4](https://github.com/rodneyrehm/viewport-units-buggyfill/releases/tag/v0.5.4)

**Comment by dillonlucente** [comment]
No problem, hope it works out.

-- 
Dillon Lucente
WILL Interactive

On July 15, 2015 at 12:31:44 PM, Rodney Rehm (notifications@github.com) wrote:

thank you for the fix, it's released as v0.5.4

**Comment by renholm** [comment]
Great work, thanks! @dillonlucente @rodneyrehm
